core/gate.h

```diff
@@ -150,8 +144,9 @@ class Gate {
 
   const std::vector<GateHook *> &hooks() const { return hooks_; }
 
-  // Inserts hook in priority order and returns 0 on success.
-  int AddHook(GateHook *hook);
+  // Creates, initializes, and then inserts gate hook in priority order.
```

Comment:
Given that `gate` is modified by this method, perhaps it should be the last arg?

Reply:
I don't have any strong feeling either way. Do we have a convention? The Google style sheet doesn't seem to say anything about modified vs unmodified, except that `T&` requires the `const`, so that if we're modifying the thing we have to use `T*` rather than `T&`.

Reply:
It's minor, so fine with me.  The style guide does say outputs come last, though whether this is an "output" is a little ambiguous:

https://google.github.io/styleguide/cppguide.html#Function_Parameter_Ordering